### DISCUSSION TITLE
OTA FU with STM32WB and STM32duino

### DISCUSSION THREAD

**Discussion by JSzypicyn** [stm32duino/Arduino_Core_STM32#1406] [Q&A]
Hello all,

I am a newbie to OTA and generally to wireless connectivity. I have developed a few custom board with STM32F series and STM32duino before. But for my next project I would like to be able to send firmware updates to the main board over the embedded BLE on STM32WB. I have absolutely no idea how to begin, I did a google search but it just escapes my understanding. I see it is possible with standard STM32CubeIDE, but how would I go about it with ArduinoIDE + STM32duino? I would be forever grateful if someone could dumb it down for me and guide me through a process required to establish OTA FU capability with STM32WB chips/STM32duino/ArduinoIDE. For testing I have acquired the STM32WB55 Nucleo Pack.

Many thanks, and if I can I will repay with a virtual beer ;)

J.

**Comment by fpistm** [comment]
Hi @JSzypicyn 
As far as I can tell is that this should be a service managed by the STM32duino library which is a fork of the ArduinoBLE.
It already exists a request about this on the ArduinoBLE:
https://github.com/arduino-libraries/ArduinoBLE/issues/180

**Reply by JSzypicyn** [reply]
Hi @fpistm ,

I see, so as it stands, it is a waiting game for the functionality to be implemented in STM32duino library? Otherwise it has to be done "natively" in STM32Cube environment with all the bootloader memory reshuffling and whatnot?

Thanks,
J.

**Reply by fpistm** [reply]
> I see, so as it stands, it is a waiting game for the functionality to be implemented in STM32duino library?

Not really. STM32duino library is a fork to add some HCI transport layer and is synced to official library regularly. I guess OTA is more high level services. So if it is provided in Arduino BLE, STM32duinoBLE will inherit it.

About Cube env I don't know. I must admit not know this part.
